custom_components/tuya_local/devices/hotshot2000_heater.yaml

```diff
@@ -0,0 +1,72 @@
+name: HOT SHOT 2000 WIFI
+products:
+  - id: lgn0xaegjbodrjax
+    name: EUROM Hot-Shot 2000 WiFi
+primary_entity:
+  entity: climate
+  dps:
+    - id: 1
+      type: boolean
+      name: hvac_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          constraint: work_mode
+          conditions:
+            - dps_val: level1
+              value: fan_only
+            - dps_val: level2
+              value: heat
+            - dps_val: level3
+              value: heat
+    - id: 2
+      type: integer
+      name: temperature
+      range:
+        min: 5
+        max: 37
+    - id: 3
+      type: integer
+      name: current_temperature
+    - id: 5
+      name: work_mode
+      type: string
+      hidden: true
+    - id: 6
+      name: preset_mode
+      type: boolean
+      mapping:
+        - dps_val: false
+          constraint: work_mode
+          conditions:
+            - dps_val: "level2"
+              value: comfort
+            - dps_val: "level3"
+              value: boost
+        - dps_val: true
+          value: eco
+    - id: 8
+      type: boolean
+      name: swing_mode
+      mapping:
+        - dps_val: false
+          value: "off"
+        - dps_val: true
+          value: "on"
+secondary_entities:
+  - entity: switch
+    name: Light
+    category: diagnostic
```

Comment:
Should be config, not diagnostic, as switch is a modifiable entity (you should see warnings in the log about this).

Reply:
How would you call the red ring below (it can be blue, green and red, indicating level1, level2 and level3 values correspondingly)?

![](https://cdn.hornbach.nl/data/shop/D04/001/780/492/163/439/DV_8_10138727_07_4c_AT_20211001051656.jpg)

Reply:
That is an indicator, maybe "Mode indicator"